src/components/ModalConfirmDelete.js

```diff
@@ -0,0 +1,48 @@
+import Modal from "./Modal.js";
+
+export default class ModalConfirmDelete extends Modal {
+  constructor(modalSelector) {
+    super(modalSelector);
+    this._modalForm = this._modalElement.querySelector(".modal__form");
+    this._modalSubmitButton = this._modalElement.querySelector(
+      ".modal__button-confirm"
+    );
+  }
+
+  //Public method setDeleteConfirmation()
+  setDeleteConfirmation(handleFormSubmit) {
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  //Public method setEventListeners()
+  setEventListeners() {
+    this._modalForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      this._handleFormSubmit();
+      this._modalForm.reset();
+      this.close();
```

Comment:
You need to **reset** the form only  in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.